### PR TITLE
Propagate multicast routes in MLDv2 reports

### PR DESCRIPTION
**Pull request by kderda**
**Description**
This PR introduces a new config option that enables appending multicast routing entries to MLDv2 reports and sending unsolicited reports on multicast route addition/deletion.

**Rationale**
Multicast routes let us add forwarding for specific prefixes or full IPv6 addresses. In case of 128-bit prefixes such addresses entries can be added to MLDv2 reports. This let us avoid joining a given multicast group by other interfaces that will forward such messages as it would lead to processing these packets by themselves (which would lead to unexpected behavior, e.g. responding to Ping Requests by forwarders) and also require additional memory used only for this purpose.

### CODE REVIEW

File: subsys/net/ip/Kconfig [hunk 1]
```diff
@@ -426,6 +426,14 @@ config NET_MAX_MCAST_ROUTES
 	  This determines how many entries can be stored in multicast
 	  routing table.
 
+config NET_MCAST_ROUTE_MLD_REPORTS
+	bool "Report multicast routes as a part of MLDv2 reports"
+	depends on ((NET_ROUTE_MCAST) && (NET_IPV6_MLD))
+	default n
```
Comment (pdgendt) [inline subsys/net/ip/Kconfig:432]: Omit `default n`, and preferably split dependency lines.

```suggestion
	depends on NET_ROUTE_MCAST
	depends on NET_IPV6_MLD
```

File: subsys/net/ip/route.c [hunk 2]
```diff
@@ -729,6 +729,42 @@ int net_route_foreach(net_route_cb_t cb, void *user_data)
...
+	/* Apply only for complete addresses */
+	if (route->prefix_len == 128) {
+		mld_event.addr = &route->group;
+		mld_event.mode = route_added ? NET_IPV6_MLDv2_MODE_IS_EXCLUDE :
+					       NET_IPV6_MLDv2_MODE_IS_INCLUDE;
+
+		net_if_foreach(send_mld_event, &mld_event);
+	}
+}
+#endif
```
Comment (rlubos) [inline subsys/net/ip/route.c:766]: I'd add `#else` here and add a dummy `#define propagate_mld_event(...)`. That way it won't be needed to add ifdefs in the code below.

**Review by rlubos — DISMISSED** [review]
Just one suggestion

**Comment by jukkar** [comment]
@konradderda could you fix the merge conflict and address the comment please.

**Comment by konradderda** [comment]
@jukkar I have applied the comment and resolved all conflicts

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**